server/src/main/java/com/cloud/api/query/dao/HostJoinDaoImpl.java

```diff
@@ -125,7 +127,11 @@ public HostResponse newHostResponse(HostJoinVO host, EnumSet<HostDetails> detail
         hostResponse.setCpuNumber(host.getCpus());
         hostResponse.setZoneId(host.getZoneUuid());
         hostResponse.setDisconnectedOn(host.getDisconnectedOn());
-        hostResponse.setHypervisor(host.getHypervisorType());
+        if (host.getHypervisorType() != null) {
+            String hypervisorType = host.getHypervisorType() != Hypervisor.HypervisorType.Custom ?
+                    host.getHypervisorType().toString() : HypervisorGuru.HypervisorCustomDisplayName.value();
```

Comment:
this should be a method of `Hypervisor.hypervisorType`

Reply:
Nice, thanks